CONTRIBUTING.md

```diff
@@ -0,0 +1,713 @@
+# Contributing
+
+This guide have some instructions and tips on how to create a new Tachiyomi extension. Please **read
+it carefully** if you're a new contributor or don't have any experience on the required languages 
+and knowledges.
+
+This guide is not definitive and it's being updated over time. If you find any issue on it, feel 
+free to report it through a [Meta Issue](https://github.com/keiyoushi/extensions-source/issues/new?assignees=&labels=Meta+request&template=06_request_meta.yml) 
+or fixing it directly by submitting a Pull Request.
+
+## Table of Contents
+
+1. [Prerequisites](#prerequisites)
+   1. [Tools](#tools)
+   2. [Cloning the repository](#cloning-the-repository)
+2. [Getting help](#getting-help)
+3. [Writing an extension](#writing-an-extension)
+   1. [Setting up a new Gradle module](#setting-up-a-new-gradle-module)
+   2. [Core dependencies](#core-dependencies)
+   3. [Extension main class](#extension-main-class)
+   4. [Extension call flow](#extension-call-flow)
+   5. [Misc notes](#misc-notes)
+   6. [Advanced extension features](#advanced-extension-features)
+4. [Running](#running)
+5. [Debugging](#debugging)
+   1. [Android Debugger](#android-debugger)
+   2. [Logs](#logs)
+   3. [Inspecting network calls](#inspecting-network-calls)
+   4. [Using external network inspecting tools](#using-external-network-inspecting-tools)
+6. [Building](#building)
+7. [Submitting the changes](#submitting-the-changes)
+   1. [Pull Request checklist](#pull-request-checklist)
+
+## Prerequisites
+
+Before you start, please note that the ability to use following technologies is **required** and 
+that existing contributors will not actively teach them to you.
+
+- Basic [Android development](https://developer.android.com/)
+- [Kotlin](https://kotlinlang.org/)
+- Web scraping
+    - [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML)
+    - [CSS selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors)
+    - [OkHttp](https://square.github.io/okhttp/)
+    - [JSoup](https://jsoup.org/)
+
+### Tools
+
+- [Android Studio](https://developer.android.com/studio)
+- Emulator or phone with developer options enabled and a recent version of Tachiyomi installed
+- [Icon Generator](https://as280093.github.io/AndroidAssetStudio/icons-launcher.html)
+
+### Cloning the repository
+
+Some alternative steps can be followed to ignore "repo" branch and skip unrelated sources, which 
+will make it faster to pull, navigate and build. This will also reduce disk usage and network traffic.
+
+<details><summary>Steps</summary>
+
+1. Make sure to delete "repo" branch in your fork. You may also want to disable Actions in the repo
+settings.
+
+    **Also make sure you are using the latest version of Git as many commands used here are pretty new.**
+
+2. Do a partial clone.
```

Comment:
is this section even needed now that apks are in separate repo?

Reply:
Might still be useful for only fetching/loading a few specific extensions into the IDE, though the point about not cloning `repo` is probably moot, yeah.

Editing `settings.gradle.kts` is another option, but people might accidentally check in their own settings.